### ISSUE TITLE
Calculating Metric other than accuracy on validation data

### ISSUE THREAD

**Issue by aritraghsh09** [tflearn/tflearn#1036]
I am using the RMSprop optimizer and the metric R2 as shown below:-

`network = regression(network, optimizer='RMSprop',loss='mean_square',metric='R2',learning_rate=0.001)`

Then I am calling fit as follows:-

`model.fit(X, Y, n_epoch=10, validation_set=(X_test,Y_test), shuffle=True, show_metric=True, batch_size=64, snapshot_step=None, snapshot_epoch=True, run_id='gal_net_r0')`

This results in output lines like this:-

`Training Step: 60 | total loss: 8.47987 | time: 1.726ss

RMSProp | epoch: 004 | loss: 8.47987 - R2: 0.6595 | val_loss: 7.42683 - val_acc: 0.6743 -- iter: 900/900`

I am not sure what is happening here. Is R2 being calculated on the validation data and being printed as val_acc? or is val_acc is actually the metric:'accuracy' -- if that is the case, then how is one supposed to calculate metrics other than accuracy on the validation data?
